File: Management/Models/CanvasModels/Assignments/CanvasTurnitinSettings.cs
namespace CanvasModel.Assignments;

public record CanvasTurnitinSettings
(
  [property: JsonPropertyName("originality_report_visibility")]
  string OriginalityReportVisibility,

  [property: JsonPropertyName("s_paper_check")]
  bool SPaperCheck,

  [property: JsonPropertyName("internet_check")]
  bool InternetCheck,

  [property: JsonPropertyName("journal_check")]
  bool JournalCheck,

  [property: JsonPropertyName("exclude_biblio")]
  bool ExcludeBiblio,

  [property: JsonPropertyName("exclude_quoted")]
  bool ExcludeQuoted,

  [property: JsonPropertyName("exclude_small_matches_type")]
  bool? ExcludeSmallMatchesType = null,

  [property: JsonPropertyName("exclude_small_matches_value")]
  uint? ExcludeSmallMatchesValue = null
);

File: nextjs/src/models/canvas/assignments/canvasTurnitinSettings.ts
export interface CanvasTurnitinSettings {
  originality_report_visibility: string;
  s_paper_check: boolean;
  internet_check: boolean;
  journal_check: boolean;
  exclude_biblio: boolean;
  exclude_quoted: boolean;
  exclude_small_matches_type?: boolean;
  exclude_small_matches_value?: number;
}
